### PR TITLE
[MXNET-345] Hardmax operator

### PR DESCRIPTION
**Pull request by ChaiBapchya**
## Description ##
Hardmax operator: python front-end api with unittest

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [X] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [X] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [X] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by ankkhedia** [comment]
@ChaiBapchya Thanks for your contribution!

@mxnet-label-bot [pr-awaiting-response, operator]

**Comment by anirudhacharya** [comment]
@apeforest please take a look.

**Comment by ChaiBapchya** [comment]
@mxnet-label-bot add [pr-work-in-progress]

**Comment by stu1130** [comment]
@mxnet-label-bot remove [pr-awaiting-response]

File: python/mxnet/ndarray/contrib.py [hunk 1]
```diff
@@ -542,3 +542,21 @@ def isnan(data):
...
+
+    Parameters
+    ----------
+    input : NDArray
+        An N-D NDArray.
+
+    Returns
+    -------
+    output: NDArray
+        The output NDarray, is a batched one-hot tensor.
```
Comment (eric-haibin-lin) [inline python/mxnet/ndarray/contrib.py]: Could you add an example?

File: python/mxnet/ndarray/contrib.py [hunk 2]
```diff
@@ -542,3 +542,31 @@ def isnan(data):
     <NDArray 2 @cpu(0)>
     """
     return data != data
+
+def hardmax(data):
```
Comment (anirudhacharya) [inline python/mxnet/ndarray/contrib.py]: why is this operator in contrib?

Comment (ChaiBapchya) [reply python/mxnet/ndarray/contrib.py]: @eric-haibin-lin suggested the same

**Comment by ChaiBapchya** [comment]
Ya, given our discussion, to implement it in front-end temporarily. Have started working on its backend implementation side-by-side.

File: tests/python/unittest/test_ndarray.py [hunk 3]
```diff
@@ -1548,6 +1548,15 @@ def test_ndarray_is_nan():
     np.testing.assert_equal(output.asnumpy(), expected_output.astype(int))
     # astype since numpy functions default return type is boolean array instead of int
 
+@with_seed()
+def test_ndarray_hardmax():
+    data = mxnet.test_utils.rand_ndarray([2,3,4],'default')
```
Comment (anirudhacharya) [inline tests/python/unittest/test_ndarray.py]: can you get a random shape using these util functions - https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/test_utils.py#L402

Comment (anirudhacharya) [reply tests/python/unittest/test_ndarray.py]: you can use `rand_shape_nd`

Comment (ChaiBapchya) [reply tests/python/unittest/test_ndarray.py]: Done

**Comment by Roshrini** [comment]
@eric-haibin-lin Can you check take review/merge this PR?
@mxnet-label-bot Update [pr-awaiting-response]

**Comment by apeforest** [comment]
LGTM. I think we can merge this frontend implementation first so ONNX can leverage this. If there is a strong demand of this operator. We can implement it in backend and move it from contrib to regular operators.

**Comment by apeforest** [comment]
@mxnet-label-bot Update [pr-awaiting-merge]

**Comment by Roshrini** [comment]
@eric-haibin-lin Can you review/merge this PR?

**Comment by vandanavk** [comment]
Will this be added in symbol/contrib.py too?

**Comment by Roshrini** [comment]
@ChaiBapchya Thanks for working on this. Can you add this for symbol as well?

**Comment by stu1130** [comment]
@mxnet-label-bot update [pr-work-in-progress]
I assume 👍 mean yes! 💯

**Comment by ChaiBapchya** [comment]
Apologies for being less explicit. Yes. I meant yes.

**Comment by pinaraws** [comment]
@ChaiBapchya Could you please fix the conflicts and retrigger the tests?

**Comment by nswamy** [comment]
This is not [per spec](https://github.com/onnx/onnx/blob/master/docs/Operators.md#Hardmax), you are always using the last axis. I am going to attempt implementing this on the backend

**Comment by piyushghai** [comment]
@nswamy Can this PR be closed if it's not as per the spec ?

**Comment by Roshrini** [comment]
@nswamy Are you working on this? If not, @ChaiBapchya Can you fix the implementation?

**Comment by roywei** [comment]
@mxnet-label-bot add [pr-awaiting-response]

**Comment by pinaraws** [comment]
@nswamy @sandeep-krishnamurthy @anirudh2290 - Please consider closing this PR since there is no follow up from the author since January.

**Comment by Roshrini** [comment]
@ChaiBapchya Closing this PR for now. Please reopen when you get time.

**Pull request closed without merging**